### PR TITLE
Add documentation for SQLite upgrade on AmazonLinux2 and CentOS

### PR DESCRIPTION
**Pull request by subashcanapathy**
Instructions for AmazonLinux2 and Centos users to upgrade SQLite3 to comply with the updated expectation from Airflow to use versions > v3.15.
The code snippets in the documentation was tested on amazonlinux:2.0 container image

related: #14208
---

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
some conflicts to solve @subashcanapathy !

**Comment by subashcanapathy** [comment]
@potiuk - resolved conflicts. waiting on the CI build

File: docs/apache-airflow/howto/set-up-database.rst [hunk 1]
```diff
@@ -103,6 +103,48 @@ An example URI for the sqlite database:
 
     sqlite:////home/airflow/airflow.db
 
+**Upgrading SQLite on AmazonLinux AMI or Container Image**
+
+AmazonLinux SQLite can only be upgraded to v3.7 using the source repos. Airflow requires v3.15 or higher. Use the
+following instructions to setup the base image (or AMI) with latest SQLite3
+
+Pre-req: You will need `wget, tar, gzip, gcc, make,` and `expect` to get the upgrade process working.
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
Pre-req: You will need ``wget``, ``tar``, ``gzip``,`` gcc``, ``make``, and ``expect`` to get the upgrade process working.
```

File: docs/apache-airflow/howto/set-up-database.rst [hunk 2]
```diff
@@ -103,6 +103,48 @@ An example URI for the sqlite database:
...
+        -DSQLITE_TEMP_STORE=3 \
+        -DSQLITE_USE_URI \
+        -O2 \
+        -fPIC"
+    export PREFIX="/usr/local"
+    LIBS="-lm" ./configure --disable-tcl --enable-shared --enable-tempstore=always --prefix="$PREFIX"
+    make
+    make install
+
+Post install add /usr/local/lib to library path
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
Post install add ``/usr/local/lib`` to library path
```

File: docs/apache-airflow/howto/set-up-database.rst [hunk 3]
```diff
@@ -103,6 +103,48 @@ An example URI for the sqlite database:
...
+AmazonLinux SQLite can only be upgraded to v3.7 using the source repos. Airflow requires v3.15 or higher. Use the
+following instructions to setup the base image (or AMI) with latest SQLite3
+
+Pre-req: You will need ``wget``, ``tar``, ``gzip``,`` gcc``, ``make``, and ``expect`` to get the upgrade process working.
+
+.. code-block:: bash
+
+  yum -y install wget tar gzip gcc make expect
+
+Download source from sqlite.org, make and install locally.
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
Download source from https://sqlite.org/, make and install locally.
```

**Review by kaxil — APPROVED** [review]
Added and committed some suggestions to your PR -- let's wait for CI to pass and we can merge it

**Comment by subashcanapathy** [comment]
Looks like the linter wants the whole word spelled out. "pre-requisite"
committing the change and trying again

**Pull request merged**